File: process/Python/data/filename.py

from json import dumps as json_dumps
import hashlib
from process.Python import TEST_RUN

def create_hash_filename(params, test_flag: bool = TEST_RUN, filename_suffix: str = None):

    if test_flag:
        return "test"

    # 1. Consistent JSON string
    json_str = json_dumps(params, sort_keys=True).encode('utf-8')
    
    # 2. Generate MD5 hash (unique signature of this dict)
    signature = hashlib.md5(json_str).hexdigest()

    if filename_suffix is not None:
        signature = f"{signature}_{filename_suffix}"

    return signature

File: process/R/data/filename.R


create_hash_filename <- function(params, test_flag = FALSE, filename_suffix = NULL) {
  
  if (test_flag) {
    return ("test")
  }
  
  if (is.list(params) && !is.null(names(params))) {
    params <- params[order(names(params))]
  }
  
  # auto_unbox=TRUE ensures scalars are not wrapped in brackets (e.g., 1 vs [1])
  json_str <- toJSON(params, auto_unbox = TRUE, digits = NA)
  
  # 2. Generate MD5 hash
  signature <- digest(json_str, algo = "md5", serialize = FALSE)
  
  if (! is.null(filename_suffix)) {
    signature <- paste0(signature, "_", filename_suffix)
  }
  
  return(signature)
}